python_modules/dagster/dagster/_core/definitions/repository_definition.py

```diff
@@ -1272,6 +1289,10 @@ def has_sensor_def(self, name: str) -> bool:
     def source_assets_by_key(self) -> Dict[AssetKey, SourceAsset]:
         return self._repository_data.get_source_assets_by_key()
 
+    @property
+    def assets_by_key(self) -> Dict[AssetKey, "AssetsDefinition"]:
```

Comment:
Doesn't roll off the tongue quit as well, but I think it would be more clear to call this `_assets_defs_by_key`.  Also, prefer `Mapping` over `Dict`.